src/lib/model/ng-artifacts-factory.ts

```diff
@@ -13,26 +13,28 @@ export class NgArtifactsFactory {
   }
 
   public calculateArtifactPathsForBuild(ngPkg: NgPackageData): NgArtifacts {
-    const pathFromRoot: string = path.resolve(ngPkg.buildDirectory, ngPkg.pathOffsetFromSourceRoot);
+    const { buildDirectory, pathOffsetFromSourceRoot, flatModuleFileName } = ngPkg;
+    const pathFromRoot: string = path.resolve(buildDirectory, pathOffsetFromSourceRoot);
 
     return {
-      main: path.join(ngPkg.buildDirectory,'bundles', this._makeUmdPackageName(ngPkg)),
-      module: path.join(ngPkg.buildDirectory, ngPkg.fullPackageName + '.es5.js'),
-      es2015: path.join(ngPkg.buildDirectory, ngPkg.fullPackageName + '.js'),
-      typings: path.join(pathFromRoot, ngPkg.flatModuleFileName + '.d.ts'),
-      metadata: path.join(pathFromRoot, ngPkg.flatModuleFileName + '.metadata.json')
+      main: path.join(buildDirectory, 'bundles', this._makeUmdPackageName(ngPkg)),
+      module: path.join(buildDirectory, 'esm5', `${flatModuleFileName}.js`),
+      es2015: path.join(buildDirectory, 'esm2015', `${flatModuleFileName}.js`),
```

Comment:
Do we have potentiaal file names conflicts here?

https://github.com/alan-agius4/ng-packagr/blob/c6f6dfb8c2cd17f3c1803d6431b9494570983594/src/lib/model/ng-package-data.ts#L65-L67

What if `@foo/bar/testing`, `@foo/bar/sub/testing` both have the "flatModuleFileName": "testing"

Reply:
won't in that case `buildDirectory` be `@foo/bar/sub`?

so i assume it should be 

```
  --esm2015
  ----testing.js
  ----sub
  ------testing.js
```

Reply:
Yes indeed, I am thinking how we can address that.

Reply:
Let's take the "full" package name excluding the npm org. E.g.:

`@foo/bar/sub/testing` should be `bar-sub-testing.js`
`@foo/bar/testing` should be `bar-testing.js`

Reply:
Agreed, unless you want to do like angular are doing something like;
ie; create a subfolder, with the library name if the file already exists.

![image](https://user-images.githubusercontent.com/17563226/33270025-eecfba74-d382-11e7-8c39-95c71bf136e4.png)

Reply:
Ok! Let's take the subfolder then!

Reply:
I did the change, let me know what you think. I still need to add some unit tests, but i'll do these later today or tomorrow :)

Reply:
Tests added